app/services/create_newsletter_signup.rb

```diff
@@ -0,0 +1,50 @@
+require 'sendgrid-ruby'
+include SendGrid
+
+class CreateNewsletterSignup
+  attr_accessor :first_name, :last_name, :email
+
+  def initialize(first_name:, last_name:, email:)
+    @first_name = first_name
+    @last_name = last_name
+    @email = email
+  end
+
+  def perform
+    sendgrid_request = request_signup
+
+    ServiceResponse.new(
+      message: "Request successfully requested!",
+      object: email,
+      success: true
+    )
+    ServiceResponse.new(
+      message: "There was a problem with your request.",
+      object: email,
+      success: false
+    )
+  end
+
+  private
+
+  def request_signup
+    data = JSON.parse('{
+      "list_ids": [
+        "'+ENV['newsletter_list_id']+'"
+      ],
+      "contacts": [
+        {
+          "first_name": "'+@first_name+'",
+          "last_name": "'+@last_name+'",
+          "email": "'+@email+'"
+        }
+      ]
+    }')
```

Comment:
You can actually build a normal rails hash here, and then just call 'to_json'. It's way easier.